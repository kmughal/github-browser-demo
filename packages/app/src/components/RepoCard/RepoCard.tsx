import React from 'react';
import { GithubRepo } from '../../stores';


type RepoCardProps = {
    addVisibilityHandler: (repoId: number) => void;
    data: GithubRepo;
};

export const RepoCard = ({ addVisibilityHandler, data }: RepoCardProps) => {
    const bgClass = data.visibility ? "bg-slate-200" : "bg-white";

    return <div
        data-testid={`repo-card-${data.id}`}
        className={`repo-card flex-col p-5 border-1 rounded-lg ${bgClass} border-purple-900 m-5 hover:bg-slate-400`}
        key={data.id}
        onClick={() => addVisibilityHandler(data.id)}
    >
        <h1 className="text-4xl flex justify-between cursor-pointer">
            <div className="flex justify-between w-full">

                <div className="flex items-center">
                    <img
                        src={data.avatarUrl}
                        alt="avatar"
                        className="mr-5 w-10 h-10 rounded-full"
                    />
                    <div className="flex flex-col">
                        <h1 className="text-lg font-bold">{data.name}</h1>
                        <p className="text-sm text-gray-600">{data.description}</p>
                    </div>
                </div>
                <div className="flex">
                    <span className="text-xl">{data.visibility ? "-" : "+"}</span>
                </div>
            </div>
        </h1>
    </div>
};