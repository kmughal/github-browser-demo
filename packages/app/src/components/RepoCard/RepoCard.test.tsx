import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { RepoCard } from './RepoCard';

describe('RepoCard Tests', () => {
    const mockRepoData = {
        id: 1,
        name: 'test',
        description: 'test description',
        visibility: true,
        avatarUrl: 'http://some.url'
    };

    const setup = () => render(
        <RepoCard
            addVisibilityHandler={jest.fn()}
            data={mockRepoData}
        />
    );

    it('should render RepoCard component', () => {
        const tree = renderer
            .create(<RepoCard
                addVisibilityHandler={jest.fn()}
                data={mockRepoData}
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should display avatar image', () => {
        setup();
        const avatarImage = screen.getByAltText('avatar');
        expect(avatarImage).toBeInTheDocument();
        expect(avatarImage).toHaveAttribute('src', mockRepoData.avatarUrl);
    });

    it('should display repo name', () => {
        setup();
        const repoNameElement = screen.getByText(mockRepoData.name);
        expect(repoNameElement).toBeInTheDocument();
    });

    it('should display repo description', () => {
        setup();
        const repoDescriptionElement = screen.getByText(mockRepoData.description);
        expect(repoDescriptionElement).toBeInTheDocument();
    });

    it('should display visibility toggle button', () => {
        setup();
        const visibilityButton = screen.getByText(mockRepoData.visibility ? '-' : '+');
        expect(visibilityButton).toBeInTheDocument();
    });
});
