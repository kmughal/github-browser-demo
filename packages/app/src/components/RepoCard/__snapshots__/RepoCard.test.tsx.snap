// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCard Tests should render RepoCard component 1`] = `
<div
  className="repo-card flex-col p-5 border-1 rounded-lg bg-slate-200 border-purple-900 m-5 hover:bg-slate-400"
  data-testid="repo-card-1"
  onClick={[Function]}
>
  <h1
    className="text-4xl flex justify-between cursor-pointer"
  >
    <div
      className="flex justify-between w-full"
    >
      <div
        className="flex items-center"
      >
        <img
          alt="avatar"
          className="mr-5 w-10 h-10 rounded-full"
          src="http://some.url"
        />
        <div
          className="flex flex-col"
        >
          <h1
            className="text-lg font-bold"
          >
            test
          </h1>
          <p
            className="text-sm text-gray-600"
          >
            test description
          </p>
        </div>
      </div>
      <div
        className="flex"
      >
        <span
          className="text-xl"
        >
          -
        </span>
      </div>
    </div>
  </h1>
</div>
`;
